src/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant/PostConfigureClientCredentialsOptions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Extensions.Options;
+
+namespace GSoft.Extensions.Http.Authentication.ClientCredentialsGrant;
+
+internal sealed class PostConfigureClientCredentialsOptions : IPostConfigureOptions<ClientCredentialsOptions>
+{
+    public void PostConfigure(string name, ClientCredentialsOptions options)
+    {
+        if (string.IsNullOrEmpty(options.CacheKey))
+        {
+            options.CacheKey = ComputeCacheKey(options);
+        }
+    }
+
+    private static string ComputeCacheKey(ClientCredentialsOptions options)
+    {
+        string hashedScopeHex;
+
+        using (var sha = SHA256.Create())
+        {
+            var hashedScopeBytes = sha.ComputeHash(Encoding.UTF8.GetBytes(options.Scope));
+            hashedScopeHex = BitConverter.ToString(hashedScopeBytes).Replace("-", string.Empty).ToLowerInvariant();
+        }
+
+        // Appending the hashed scope to the cache key may result in shorter cache keys compared to appending the raw scope
+        return $"GSoft.Authentication.ClientCredentialsGrant.{options.ClientId}.{hashedScopeHex}";
```

Comment:
Are `ClientId` going to be different for every service? If not, then the risk of collisions are pretty high since most services will end up with scopes like `read`, `write`, `admin` and so on...

Reply:
Alright works for me!

Reply:
I'm reopening this conversation after talking with my teammates. The user-input `name` isn't necessary in the cache key.

It's an arbitrary string that will end up in a database primary key. `ClientId` combined with the `Scope` guarantees no collisions already. It is litterally:
* `ClientId`: the caller (so an application A)
* `Scope`: the recipent(s) + optional permissions (so an application B with optional permissions)

So the cache key represents something like "_I'm service A and I want to talk with service B with these permissions_".

There is then no need to add an arbitrary user-input `name`. `ClientId` is also user-input but it is not arbitrary, it is an identifier provided by an identity provider.

Reply:
I'll let you resolve this comment. My concerns have been addressed :)